### PR TITLE
Feature/ Add styling per mode to statusbar

### PR DESCRIPTION
**Pull request by Akin909**
statusbar

Add styling based on current vim mode and also separate line/col element in statusbar

### CODE REVIEW

**Comment by bryphe** [comment]
Awesome! I've been missing this 😄

File: src/editor/Core/Theme.re [hunk 1]
```diff
@@ -22,6 +22,20 @@ module EditorColors = {
...
+    statusBarItemNormalModeBackground: Color.t,
+    statusBarItemOperatorModeBackground: Color.t,
+    statusBarItemCommandlineModeBackground: Color.t,
+    statusBarItemVisualModeForeground: Color.t,
+    statusBarItemInsertModeForeground: Color.t,
+    statusBarItemReplaceModeForeground: Color.t,
+    statusBarItemNormalModeForeground: Color.t,
+    statusBarItemOperatorModeForeground: Color.t,
+    statusBarItemCommandlineModeForeground: Color.t,
+    statusBarItemForeground: Color.t,
```
Comment (bryphe) [inline src/editor/Core/Theme.re]: Should we just use `statusBar.background` and `statusBar.foreground` for these?

Thinking of how they relate here:
https://code.visualstudio.com/api/references/theme-color#status-bar-colors

File: src/editor/Core/Theme.re [hunk 2]
```diff
@@ -22,6 +22,20 @@ module EditorColors = {
     editorMenuForeground: Color.t,
     editorMenuItemSelected: Color.t,
     tabActiveForeground: Color.t,
+    statusBarItemVisualModeBackground: Color.t,
```
Comment (bryphe) [inline src/editor/Core/Theme.re]: Thanks for adding the mode colors and defaults! There's no real equivalent in VSCode so I think we will definitely need our own.

I'm wondering if we could generalize this to something like `oniVisualModeBackground`  (using the 'oni' prefix as a convention for theme colors that aren't available in VSCode). 

Seeing this change made me want these colors on top of the active tab (like we did in Oni1), so I was thinking there'd be a case where we might re-use these colors in other parts of the UI, too.

**Comment by Akin909** [comment]
@bryphe addressed your comments btw

**Pull request merged**